tests/unittest/CFGProgrammer/CFGProgrammer_test.cpp

```diff
@@ -893,4 +925,88 @@ TEST_F(HwDevicesTestFixture, FindDeviceByNameTest) {
   EXPECT_EQ(device.flashSize, hwDevices.getDevices()[1].flashSize);
 }
 
+class ProgramerTestFixture : public ::testing::Test {
+protected:
+  void SetUp() override {
+    cmdarg.command = "programmer";
+    cmdarg.toolPath = "build/bin/openocd";
+    original_cin = std::cin.rdbuf(input_stream.rdbuf());
+  }
+  void TearDown() override {
+    std::cin.rdbuf(original_cin);
+  }
+  CFGCommon_ARG cmdarg;
+  std::istringstream input_stream;
+  std::streambuf *original_cin = nullptr;
+};
+
+TEST_F(ProgramerTestFixture, ExecuteOTPCommandWithDummyCompiler) {
+  const char *argv[] = { "otp", "-c", "1", "bitstream.cfgbit" };
+  int argc = 4;
+
+  auto arg = std::make_shared<CFGArg_PROGRAMMER>();
+  arg->parse(argc, argv, nullptr);
+  cmdarg.arg = arg;
+  cmdarg.compilerName = "dummy";
+  programmer_entry(&cmdarg);
+
+  // just for code coverage but nothing to be observed for now
+}
+
+TEST_F(ProgramerTestFixture, ExecuteOTPCommandWithOpenFPGACompilerAndYFlag) {
+  const char *argv[] = { "otp", "-c", "1", "bitstream.cfgbit", "-y" };
+  int argc = 5;
+
+  auto arg = std::make_shared<CFGArg_PROGRAMMER>();
+  arg->parse(argc, argv, nullptr);
+  cmdarg.arg = arg;
+  cmdarg.compilerName = "openfpga";
+  programmer_entry(&cmdarg);
+
+  // just for code coverage but nothing to be observed for now
+}
 
+TEST_F(ProgramerTestFixture, ExecuteOTPCommandWithPromptingAnd_Y_Anwser) {
+  const char *argv[] = { "otp", "-c", "1", "bitstream.cfgbit" };
+  int argc = 4;
+
+  input_stream.str("Y");
+
+  auto arg = std::make_shared<CFGArg_PROGRAMMER>();
+  arg->parse(argc, argv, nullptr);
+  cmdarg.arg = arg;
+  cmdarg.compilerName = "openfpga";
+  programmer_entry(&cmdarg);
+
+  // just for code coverage but nothing to be observed for now
+}
+
+TEST_F(ProgramerTestFixture, ExecuteOTPCommandWithPromptingAnd_N_Anwser) {
```

Comment:
Typo here and other tests where this is used.
```suggestion
TEST_F(ProgramerTestFixture, ExecuteOTPCommandWithPromptingAnd_N_Answer) {
```

Reply:
These UTs are moved.